June newsletter (#584)
ropensci/roweb3: content/blog/2023-06-23-ropensci-news-digest-june-2023/index.Rmd

You can also read the post in [Spanish](/es/blog/2023/06/06/r-universe-stars-4-es/).

### Downloading snapshots and creating stable R packages repositories using R-universe!

The [new snapshot API](/blog/2023/05/31/runiverse-snapshots/) lets you download a full copy of any CRAN-like repository on R-universe. You can use such a snapshot to mirror the entire CRAN-like repository on your own servers, or for example to build a stable, validated release of your package suite.

### Statistical software review now completely integrated with standard review process

Statistical software review has finally reached a stable and mature state, with all organizational processes now entirely integrated with the standard review process.
Developers submitting software should notice little difference, except hopefully faster initial processing of submissions.
Each submission is now smoothly handled by members of our team of statistical software editors:

-   [Adam Sparks](/author/adam-sparks/)
-   [Emi Tanaka](https://emitanaka.org/)
-   [Jouni Helske](/author/jouni-helske/)

> -   [Paula Moraga](http://www.paulamoraga.com/)

Reviewer: ```suggestion
-   [Paula Moraga](https://www.paulamoraga.com/)
```

-   [Rebecca Killick](/author/rebecca-killick/)

> -   [Toby Hocking](http://tdhock.github.io/)

Reviewer: ```suggestion
-   [Toby Hocking](https://tdhock.github.io/)
```

> We current accept software in the eight categories described in our [*Statistical Software Development Guide*](https://stats-devguide.ropensci.org/standards.html), and are still working on integrating the two additional categories of statistical workflow and network software.

Reviewer: ```suggestion
We currently accept software in the eight categories described in our [*Statistical Software Development Guide*](https://stats-devguide.ropensci.org/standards.html), and are still working on integrating the two additional categories of Statistical Workflow and Network Software.
```

We encourage anybody thinking about submitting to contact any one of the editors, or the two leaders of the software review team, [Noam Ross](/author/noam-ross/) and [Mark Padgham](/author/mark-padgham/), at any time, including anybody interested in submitting software in the upcoming categories of workflow or network software.

### rOpenSci's Communication Channels for Safe and Friendly Exchange

> As [we anticipated in November 2022](/blog/2022/11/16/mastodon-en) the changes implemented by Twitter have generated a less safe and friendly space for our community.  
> That is why from _June 2023_, we will stop interacting on this platform. 

Reviewer: ```suggestion
As [we anticipated in November 2022](/blog/2022/11/16/mastodon-en) the changes implemented by Twitter have generated a less safe and friendly space for our community.
That is why from _June 2023_, we will stop interacting on this platform. 
```

We will highlight the benefits of being part of the program for you and for your community, what kind of learning, 
activities and opportunities an open source community champions program provides. 
_Yani_ will present the details of our Champion Program and answer all your question about it.

### Coworking

Join us for social coworking & office hours monthly on first Tuesdays! 
Hosted by Steffi LaZerte and various community hosts. 
Everyone welcome. 
No RSVP needed. 
Consult our [Events](/events) page to find your local time and how to join.

* Tuesday, July 4th, 14:00 European Central / 12:00 UTC ["Create/Update your 'Happy File'/'Brag Document'!"](/events/coworking-2023-07/) *Hosted by [Maëlle Salmon](/author/maëlle-salmon) and [Steffi LaZerte](/author/steffi-lazerte/)*
    - Explore what goes into a [‘Happy File’](https://twitter.com/JennyBryan/status/1582862196870373377)/[‘Brag Document’](https://jvns.ca/blog/brag-documents/) and why you need one.
    - Start collecting items to add to your ‘Happy File’.
    

> - Tuesday, August 1st, 09:00 Americas Pacific (16:00 UTC), [Spatial data in R](/events/coworking-2023-08/) *Hosted by [Mike Mahoney](/author/mike-mahoney) and [Steffi LaZerte](/author/steffi-lazerte/)*

Reviewer: ```suggestion
- Tuesday, August 1st, 09:00 Americas Pacific (16:00 UTC), ["Spatial data in R"](/events/coworking-2023-08/) *Hosted by [Mike Mahoney](/author/mike-mahoney) and [Steffi LaZerte](/author/steffi-lazerte/)*
```

### Call for comaintainers

Refer to our somewhat [recent blog post](/blog/2022/10/17/maintain-or-co-maintain-an-ropensci-package/#packages-looking-for-co-maintainers) to identify other packages where help is especially wished for!
See also our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

## Package development corner

Some useful tips for R package developers. :eyes:

### How to produce code hints for users?

Sometimes your code can infer what the next user step should be, or what the next _possible_ user _steps_ could be.
How to make it easy for the user to run these hints?

Here's some inspiration:

> * You could use [`cli::ui_todo()`](https://usethis.r-lib.org/reference/ui.html), as done in usethis. By the way if your package is using `ui_` functions from usethis, don't miss the cli article on how to [migrate your code to using cli instead](https://cli.r-lib.org/articles/usethis-ui.html), as that's what's recommended. 

Reviewer: ```suggestion
* You could use [`cli::ui_todo()`](https://usethis.r-lib.org/reference/ui.html), as done in usethis. By the way if your package is using `ui_` functions from usethis, don't miss the cli article on how to [migrate your code to using cli instead](https://cli.r-lib.org/articles/usethis-ui.html), as that's what's currently recommended. 
```

> * Also in the cli package, a special kind of links that runs the code if it is clicked on, in RStudio IDE: ["Click to run code"](https://cli.r-lib.org/reference/links.html?q=run#click-to-run-code). You might know that from running snapshot tests with testthat, and getting told to review/accept them with these special links.

Reviewer: ```suggestion
* Also in the cli package, a you can provide a special kind of link that runs the code if clicked in the RStudio IDE: ["Click to run code"](https://cli.r-lib.org/reference/links.html?q=run#click-to-run-code). You might recognize this from running snapshot tests with testthat, and receiving the message to review/accept them with these special links.
```

```r
cli::cli_text("Run {.run praise::praise()} now!")
```

> * If your code is run in RStudio IDE (checkable with [`rstudioapi::isAvailable()`](https://rstudio.github.io/rstudioapi/reference/isAvailable.html)), you could use [`rstudioapi::sendToConsole()`](https://rstudio.github.io/rstudioapi/reference/sendToConsole.html). [Example](https://github.com/cynkra/fledge/blob/520b9b2e36da7bc58136378407f1f4454bfbeadf/R/finalize-version.R#L63-L80=).

Reviewer: I'm not sure what the purpose of doing this is... does my suggestion make sense?
```suggestion
* If your code is run in RStudio IDE (checkable with [`rstudioapi::isAvailable()`](https://rstudio.github.io/rstudioapi/reference/isAvailable.html)), you could use [`rstudioapi::sendToConsole()`](https://rstudio.github.io/rstudioapi/reference/sendToConsole.html) to send example code to the console for the user to consider ([example](https://github.com/cynkra/fledge/blob/520b9b2e36da7bc58136378407f1f4454bfbeadf/R/finalize-version.R#L63-L80=)).
```
